### PR TITLE
Added support for a JSON health endpoint (rebased)

### PR DESCRIPTION
**Pull request by aleksihakli**
Added support for JSON based health checks with tests on top of the current master branch.

### CODE REVIEW

**Comment by aleksihakli** [comment]
@codingjoe I added support for the JSON responses based on the `Accept` header.

I don't know if a CBV is the best option for status rendering as we might want to customize the endpoint and add support for multiple different types of medias such as YAML etc. in the long run. Should we just use FBVs in the endpoint?

**Comment by aleksihakli** [comment]
@codingjoe I think this is ready work-wise and good for merging from my part if it looks good to you.

File: health_check/views.py [hunk 1]
```diff
@@ -19,4 +20,11 @@ def get(self, request, *args, **kwargs):
             errors += plugin.errors
         plugins.sort(key=lambda x: x.identifier())
         status_code = 500 if errors else 200
+
+        if 'application/json' in request.META.get('HTTP_ACCEPT', ''):
+            return JsonResponse(
+                {str(p.identifier()): str(p.pretty_status()) for p in plugins},
```
Comment (codingjoe) [inline health_check/views.py]: I would put this part into a different method. That way users can overwrite the structure of the dict that gets serialised into a JSON

**Review by codingjoe — CHANGES_REQUESTED** [review]
@aleksihakli I like it, just one tiny comment.
Oh and please add documentation 😉

**Comment by aleksihakli** [comment]
@codingjoe I refactored the JSON endpoint so that it can be customized easily and added docs about the JSON output and customization to README.

I guess #31 can be closed after this as well, finally :)

**Comment by codingjoe** [comment]
Released in 2.2.0

**Pull request merged**